### PR TITLE
fix and refactor dockerfiles

### PR DESCRIPTION
**Pull request by escapewindow**
- update `.dockerignore` to avoid copying `__pycache__` and `*.pyc` files
- update `README.rst`
- downgrade gnupg on the python docker images to `2.0.26-6+deb8u2`
- rename `/data` to `/builds`
- move scriptworker from `/builds/.` to `/builds/scriptworker` to keep
  things cleaner and avoid lint/pytest warnings+errors for
  non-scriptworker code (e.g. puppet)
- consolidate the `.test` and `.gnupg` dockerfiles, get all tests passing in
  the docker container
- add a py37 version. This only differs from the py36 version in the
  `FROM` line; we may want to either templatize these for less duplication
  or allow for multiple versions of python on a single image.

Both these dockerfiles are test-oriented, whether unit tests or gpg testing. I'd also like to create some that could be used for deployment.

@JohanLorenzo , @Callek , I know you both tried using the dockerfile before with varying levels of success. How does this look?

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -69,15 +69,17 @@ then
 
 It's also possible to create a ``./secrets.json`` as above, then::
 
-    cp docker/Dockerfile.test Dockerfile
-    docker build -t scriptworker-test . && docker run scriptworker-test tox
+    PY_VERSION=37  # or 36
+    cp docker/Dockerfile.test-py$PY_VERSION Dockerfile
+    docker build -t scriptworker-test . && docker run scriptworker-test tox-py$PY_VERSION
```
Comment (JohanLorenzo) [inline README.rst]: FYI: I couldn't run these instructions on my machine:

```
[...]
Using /usr/local/lib/python3.7/site-packages
Finished processing dependencies for scriptworker==16.1.0
Removing intermediate container b69422b9d3c4
 ---> 38b7ad2f4d90
Successfully built 38b7ad2f4d90
Successfully tagged scriptworker-test:latest
docker: Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"tox-py37\": executable file not found in $PATH": unknown.
ERRO[0001] error waiting for container: context canceled 
```

Comment (escapewindow) [reply README.rst]: Pushed a fix.

**Comment by escapewindow** [comment]
Awesome, thanks! I figure I'll land this as is once I deal with the intermittent travis tests... we can improve the dockerfiles incrementally from here, now that they're working again :)

File: docker/Dockerfile.test-py36 [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+COPY version.json /builds/scriptworker/
+COPY secrets.json /builds/scriptworker/
+
+# allow for rebuild_gpg_homedirs testing
+COPY docker/gnupg.yaml /builds/scriptworker/gnupg.yaml
+COPY scriptworker/test/data/gpg/keys/scriptworker@example.com.pub /builds/my_pubkey.asc
+COPY scriptworker/test/data/gpg/keys/scriptworker@example.com.sec /builds/my_privkey.asc
+# get trusted gpg pubkeys
+RUN git clone https://github.com/mozilla-releng/cot-gpg-keys.git /builds/key_repo
+RUN git clone https://github.com/mozilla-releng/build-puppet.git /builds/puppet
```
Comment (tomprince) [inline docker/Dockerfile.test-py36]: I'm curious what the use case for this is.

Comment (escapewindow) [reply docker/Dockerfile.test-py36]: We have the trusted gpg pubkeys for scriptworker cot [here](https://github.com/mozilla-releng/build-puppet/tree/master/modules/scriptworker/files/git_pubkeys). I used to curl a tarball from hg.m.o; that's not an option with github. I think I had the option of using svn to clone a subdir, or just cloning via git.

**Comment by escapewindow** [comment]
> https://docs.docker.com/engine/reference/builder/#arg allows parameterizing images.

Nice! I'll play with that to make sure the `FROM` can use it, likely in a followup.

**Comment by escapewindow** [comment]
Closing & reopening to try to get Github to realize this PR is mergeable.

**Pull request merged**